### ISSUE TITLE
[Core Port Req.] touchhle

### ISSUE THREAD

**Issue by InfamousKnight** [TASEmulators/BizHawk#3956]
### Upstream info
- [Website]https://touchhle.org/
- Target platforms: windows/linux
- [Source repo) https://github.com/touchHLE/touchHLE
- Language(s): rust
- License:  Mozilla Public License, version 2.0.

### Merits
Because of some ios exclusive titles. I would use it for driver and crystal defenders vanguard

### Technical details
[//]: # "Non-exhaustive list of things to consider:"
- (able to build .dll/.so for P/Invoke?) not easily since it doesnt have libretro yet I guess?
- (frontend/backend separation--for example, can backend be built without SDL?) duuno
- (can force single-threaded?) dunno
- (has I/O abstraction accepting byte array, or only accepts file paths? for disc-based consoles, can swap out implementation for BizHawk's?) dunno
- (savestate quality) doesnt have save states. Would have to be waterboxed

**Comment by InfamousKnight** [comment]
linux is supported: https://github.com/Oscar1640/TouchHLE-Linux-builds-UNOFFICIAL/releases

**Comment by SuperGamer18** [comment]
touchHLE is in a somewhat heavy WIP status, so I don't really think it will come to BizHawk anytime soon.

**Comment by InfamousKnight** [comment]
The same thing can be said about a lot of cores out there. Even ones that are well developed won't be coming anytime soon either.

**Comment by YoshiRulz** [comment]
I initially thought the same, but look at their game compatibility list—their 5-star list is about as long as the full library of some of our emulated systems.

**Comment by SuperGamer18** [comment]
@InfamousKnight well, yes, that is a correct statement, but you can really see that touchHLE is in a WIP state, even when compared to those well-developed cores/emulators you say.
@YoshiRulz that is also a very good point, but why even consider adding iOS when there's no Android emulation core here? (Also as you probably know, Android emulation is much more advanced than iOS emulation, yet both are far from perfect)

I'm not against adding touchHLE to BizHawk btw, I'm just saying that it doesn't make much sense to add it in the state that it currently is. Heck, I would even dare to say that obscure platforms such as the My Vision, CD-i, etc. have better emulation than iOS.

**Comment by YoshiRulz** [comment]
>> I initially thought the same, but look at their game compatibility list
>
> The thing is the compatibility list is also a small amount compared to the actual available list of iOS 2/3 titles. [...]

You've missed my point. I'm saying that its supported library *in absolute terms* is comparable to some of our other systems, and hence *I didn't immediately reject this request*.

> [comparisons to other ports and requests]

We have no policy of prioritising certain cores for inclusion—the only hard rule is that they need a certain level of functionality (rerecording-capable). Other than that, there are only technical barriers. Whatever developers choose to work on is what makes it in.

> [...] why even consider adding iOS when there's no Android emulation core here? (Also as you probably know, Android emulation is much more advanced than iOS emulation [...]

Android emulators, as far as I know, are all QEMU-based i.e. they use host virtualisation. We can't do that. Knowing nothing about touchHLE apart from that it's written in Rust, it's likely to be structured in a way that cleanly separates the emulation code from the interface, and is unlikely to have savestate issues (now or in the future—OP said it didn't have savestates at the time).

**Comment by SuperGamer18** [comment]
@YoshiRulz fair point, but I wasn't even asking for an Android emulation core anyways (AFAIK Android emulators don't have savestates either)
